client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -133,6 +133,8 @@ private Request buildQueryRequest(ClientSession session, String query)
 
         session.getTraceToken().ifPresent(token -> builder.addHeader(TRINO_HEADERS.requestTraceToken(), token));
 
+        session.getUser().ifPresent(user -> builder.header(TRINO_HEADERS.requestUser(), session.getUser().get()));
```

Comment:
this should be done in `io.trino.client.StatementClientV1#prepareRequest`

in current code the correct value is sent in initial request, but incorrect value is sent in subsequent requests.
(if the value in subsequent requests is not needed, let's remove it completely, else send the correct value)

Reply:
From what I can tell, the headers are used when setting up the initial session but once the user is authenticated the session does not check them again.

Reply:
I did rework the use in StatementClientV1 a little in the last fixup though, hopefully makes more sense.